app/questionnaire/path_finder.py

```diff
@@ -59,19 +59,32 @@ def build_path(self):
                 group_blocks = self._build_blocks_for_group(group, instance_idx)
                 blocks += group_blocks
 
-            if group['id'] in first_groups:
-                this_location = Location(group['id'], 0, group['blocks'][0]['id'])
+            first_group_instance = self._get_first_group_instance_in_block(group, blocks)
+            if first_group_instance is not None:
+                if group['id'] in first_groups:
+                    this_location = Location(group['id'], first_group_instance, group['blocks'][0]['id'])
 
-            if blocks:
-                path, block_index = self._build_path_within_group(blocks, block_index, this_location, path)
+                if blocks:
+                    path, block_index = self._build_path_within_group(blocks, block_index, this_location, path)
 
         return path
 
+    @staticmethod
+    def _get_first_group_instance_in_block(group, blocks):
+        """returns the first non skipped group instance of a block else None"""
+        ret_val = None
+        try:
+            first_block = next(block for block in blocks if block['group_id'] == group['id'])
+            ret_val = first_block['group_instance']
+        except (StopIteration, KeyError):  # Either no group instances (all skipped) or group has no 'id'
+            pass
+        return ret_val
+
     def _build_blocks_for_group(self, group, instance_idx):
         for block in group['blocks']:
             skip_conditions = block.get('skip_conditions')
             if skip_conditions and evaluate_skip_conditions(
-                    skip_conditions, self.metadata, self.answer_store):
+                    skip_conditions, self.metadata, self.answer_store, instance_idx):
```

Comment:
What happens here if we are trying to skip on an answer that isn't in a repeating group?

Reply:
group id will be zero and return the first one

Reply:
In the when group evaluation it checks if the independent answer is in a repeating group, if it is it uses the group_instance of zero . Added tests to show that